src/Tempest/Console/src/GenericConsole.php

```diff
@@ -197,17 +226,38 @@ public function progressBar(iterable $data, Closure $handler): array
         return $this->component(new ProgressBarComponent($data, $handler));
     }
 
-    public function search(string $label, Closure $search): mixed
+    public function search(string $label, Closure $search, ?string $default = null): mixed
     {
-        return $this->component(new SearchComponent($label, $search));
+        return $this->component(new SearchComponent($label, $search, $default));
     }
 
-    private function interactiveSupported(): bool
+    public function supportsTty(): bool
     {
+        if ($this->supportsTty === false) {
+            return false;
+        }
+
         if ($this->componentRenderer === null) {
             return false;
         }
 
+        if (! $this->supportsPrompting()) {
```

Comment:
So if I get it right, prompting is available for both static and dynamic components?

Reply:
Yes! Prompting is letting the user type on the terminal. Disabling prompting is useful when there is no visible terminal, so in automations for instance.